pilot/pkg/serviceregistry/kube/controller/fake.go

```diff
@@ -44,6 +44,7 @@ func (fx *FakeXdsUpdater) ConfigUpdate(req *model.PushRequest) {
 	if req != nil && len(req.ConfigsUpdated) > 0 {
 		for key := range req.ConfigsUpdated {
 			id = key.Name
+			break
```

Comment:
This is not quite correct, maybe we should send all the config updated to the Events channel. Could you debug?

Reply:
ok, i will open another PR to correct this.